lib/client/client.go

```diff
@@ -92,8 +92,8 @@ func (proxy *ProxyClient) GetActiveSessions(ctx context.Context) ([]types.Sessio
 // GetSites returns list of the "sites" (AKA teleport clusters) connected to the proxy
 // Each site is returned as an instance of its auth server
 //
-func (proxy *ProxyClient) GetSites() ([]types.Site, error) {
-	proxySession, err := proxy.Client.NewSession()
+func (proxy *ProxyClient) GetSites(ctx context.Context) ([]types.Site, error) {
+	proxySession, err := tracing.NewSSHSession(ctx, proxy.Client)
```

Comment:
rather than replacing `NewSession` with `tracing.NewSSHSession` everywhere, I wonder about creating a `TracingSSHClient` struct which embeds `ssh.Client` and overrides `NewSession` to set up the trace.

You would need to replace `*ssh.Client` with some new interface everywhere, but then we know that every `NewSession` sets up the trace and someone can't easily just add another call that forgets to set it up.

Maybe you considered it already, what do you think?